### PR TITLE
Add IPForwarding to API

### PR DESCRIPTION
**Pull request by alexander-demichev**
**What type of PR is this?**

/kind api-change

**What this PR does / why we need it**:

In the current implementation, IPForwarding field is hardcoded to true. This PR adds IPForwarding with two possible options: enabled and disabled. The reason why it's not a bool is that I'm trying to keep behavior consistent with the current implementation and not introduce breaking changes, I believe having a string field that defaults to `Enabled` is better than a bool that defaults to true when not set.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Add IPForwarding to API
```

### CODE REVIEW

File: api/v1alpha3/conversion.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+package v1alpha3
+
+import (
+	conversion "k8s.io/apimachinery/pkg/conversion"
+	v1beta1 "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+)
+
+// Convert_v1beta1_GCPMachineSpec_To_v1alpha3_GCPMachineSpec is an autogenerated conversion function.
+func Convert_v1beta1_GCPMachineSpec_To_v1alpha3_GCPMachineSpec(in *v1beta1.GCPMachineSpec, out *GCPMachineSpec, s conversion.Scope) error {
```
Comment (pydctw) [inline api/v1alpha3/conversion.go:25]: This function can move to api/v1alpha3/gcpmachine_conversion.go

File: api/v1alpha4/conversion.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+package v1alpha4
+
+import (
+	conversion "k8s.io/apimachinery/pkg/conversion"
+	v1beta1 "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+)
+
+// Convert_v1beta1_GCPMachineSpec_To_v1alpha4_GCPMachineSpec is an autogenerated conversion function.
+func Convert_v1beta1_GCPMachineSpec_To_v1alpha4_GCPMachineSpec(in *v1beta1.GCPMachineSpec, out *GCPMachineSpec, s conversion.Scope) error {
```
Comment (pydctw) [inline api/v1alpha4/conversion.go:25]: Same as v1alpha3. Could we move this function to api/v1alpha4/gcpmachine_conversion.go?

**Comment by pydctw** [comment]
@alexander-demichev, thanks for the PR. Looks good with minor comments

**Comment by alexander-demichev** [comment]
@pydctw all fixed, thanks for review.

**Comment by pydctw** [comment]
Looks good. Could you squash the commits before merging?

**Comment by cpanato** [comment]
> Looks good. Could you squash the commits before merging?

added the label to prow squash the commits when merging :)

**Pull request merged**